pyogrio/raw.py

```diff
@@ -103,21 +103,7 @@ def read(
             "geometry": "<geometry type>"
         }
     """
-    if hasattr(path_or_buffer, "read"):
-        path_or_buffer = path_or_buffer.read()
-
-    from_buffer = False
-    if isinstance(path_or_buffer, bytes):
-        from_buffer = True
-        ext = ""
-        is_zipped = path_or_buffer[:4].startswith(b'PK\x03\x04')
-        if is_zipped:
-            ext = ".zip"
-        path = buffer_to_virtual_file(path_or_buffer, ext=ext)
-        if is_zipped:
-            path = "/vsizip/" + path
-    else:
-        path = vsi_path(str(path_or_buffer))
+    path, from_buffer = get_vsi_path(path_or_buffer)
```

Comment:
For some bizarre reason, this gives a failure in reading an in-memory GPKG file (see failing CI), although the helper function contains _literally_ the same code as it was originally inline here.

Reply:
I'm not spotting anything that should cause this, and unhelpfully, the tests all pass locally (MacOS).

It's almost like we are seeing interaction between test cases where by having a greater prevalence of calling `remove_virtual_file` across tests, that they are interferring with the state of subsequent tests.

Reply:
> unhelpfully, the tests all pass locally (MacOS).

They also pass on CI for MacOS ..

Reply:
I can reproduce locally using the GDAL 3.5 docker container.  It also fails for shapefile if I add that to the test case.

Interestingly, it also fails if I paste the code removed from raw.py back in so that is used instead.